boutvecma/encoder.py

```diff
@@ -1,7 +1,47 @@
+import os
+from copy import deepcopy
 from easyvvuq.encoders import BaseEncoder
+from boutdata.data import BoutOptions, BoutOptionsFile
+
 
 class BOUTEncoder(BaseEncoder, encoder_name="bout++_encoder"):
+    def __init__(self, template_input=None):
+        """Read an existing BOUT.inp file to use as a template.
+        If no input is given, an empty set of options will be created
+
+        Example
+        -------
+
+        from boutvecma.encoder import BOUTEncoder
+        encoder = BOUTEncoder("data/BOUT.inp")
+        """
+        if template_input:
+            self._options = BoutOptionsFile(template_input)
+        else:
+            self._options = BoutOptions()
+
+    def encode(self, params={}, target_dir=""):
```

Comment:
I checked the actual source code, they do use the better `params=None` default argument

```suggestion
    def encode(self, params=None, target_dir=""):
```

and then `if params is None: params = {}` later on.

Reply:
Thanks @ZedThree this is done